app/models/user.rb

```diff
@@ -3,4 +3,13 @@ class User < ApplicationRecord
   has_many :events, through: :attendances , dependent: :destroy
 
   validates :name, presence: true, uniqueness: true
+
+
+  def self.get_upcoming(user)
+    user.events.where("date > ?",Time.zone.now)
+  end
+
+  def self.get_previous(user)
+    user.events.where("date < ?",Time.zone.now)
+  end
```

Comment:
What's the point of using self and an argument? It's redundant
You can do something like
```ruby
  def get_upcoming
    events.where("date > ?",Time.zone.now)
  end
```